cli/create.go

```diff
@@ -239,6 +239,10 @@ func createSandbox(ociSpec oci.CompatOCISpec, runtimeConfig oci.RuntimeConfig,
 		return vc.Process{}, fmt.Errorf("BUG: Container list from sandbox is wrong, expecting only one container, found %d containers", len(containers))
 	}
 
+	if err := addContainerIDMapping(containerID, sandbox.ID()); err != nil {
```

Comment:
I think we talked that this could be done in virtcontainers itself. Any reason this is being done in the CLI?

Reply:
Well I was about to say that we were still needing a lock, but now I am confused. I think this could work:
- `fetchSandboxIDs(containerID)` returns a list of what is inside the directory `/.../containerID/`. No lock needed since the call into `readdir()` is atomic.
- `addSandboxID(containerID, sandboxID)` would add a new sandbox using `os,MkdirAll()`, meaning the path `/.../containerID/sandboxID` would be created, no matter if the parent directory `/.../containerID` exist or not.
- `delSandboxID(container, sandboxID)` would remove the path `/.../containerID/sandboxID` and after, it would call into `os.Rmdir()` `/.../containerID` without checking the error. Indeed, the call to rmdir might fail if something is still in the directory, and also if a `addSandboxID()` has just been called.

Reply:
Ok so from our offline discussion, `os.MkdirAll()` is obviously not atomic, and that's the reason why we would need some locking. But given the fact that this case would never happen since we will not use it for real, we will never hit it. Let me ask others about what they think.